src/main/java/io/cryostat/graphql/ArchivedRecordings.java

```diff
@@ -66,6 +69,31 @@ public TargetNodes.ArchivedRecordings archived(
         return out;
     }
 
+    @NonNull
+    public ArchivedRecording doDelete(@Source ArchivedRecording recording) {
+        recordingHelper.deleteArchivedRecording(recording.jvmId(), recording.name());
+        return recording;
+    }
+
+    @NonNull
+    public ArchivedRecording doPutMetadata(
+            @Source ArchivedRecording recording, MetadataLabels metadataInput) {
+        recordingHelper.updateArchivedRecordingMetadata(
+                recording.jvmId(), recording.name(), metadataInput.getLabels());
+
+        String downloadUrl = recordingHelper.downloadUrl(recording.jvmId(), recording.name());
+        String reportUrl = recordingHelper.reportUrl(recording.jvmId(), recording.name());
+
+        return new ArchivedRecording(
+                recording.jvmId(),
+                recording.name(),
+                downloadUrl,
+                reportUrl,
+                new Metadata(metadataInput.getLabels(), metadataInput.getExpiry()),
```

Comment:
This looks like a bug: the `expiry` is reported in the API response as if it has been updated, but `updateArchivedRecordingMetadata` only takes the new `labels` and not the `expiry`.

Reply:
Thinking about this API design a bit more, I guess the `expiry` doesn't make sense for the client to provide on the `doPutMetadata(ActiveRecording)` case, since active recordings don't have an expiry date at all. It *might* make sense to update the expiry date on an archived recording,

See discussion here: https://github.com/cryostatio/cryostat3/issues/269 - we aren't really using the expiry date for its intended purpose yet, so maybe we're getting ahead of ourselves in making it mutable. Mutability doesn't seem like a strongly useful feature, but it could place constraints on how we implement the file TTL.

tl;dr let's get rid of the `expiry` from the `MetadataLabels` input, at least for now.